cpp/CMakeLists.txt

```diff
@@ -802,6 +802,10 @@ install(DIRECTORY ${CUDF_SOURCE_DIR}/include/cudf ${CUDF_SOURCE_DIR}/include/cud
                   ${CUDF_SOURCE_DIR}/include/nvtext DESTINATION ${CMAKE_INSTALL_INCLUDEDIR}
 )
 
+if(dlpack_SOURCE_DIR)
+  install(DIRECTORY ${DLPACK_INCLUDE_DIR}/ DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
```

Comment:
I disagree with installing private dependencies. We don't install  other private header only dependencies such as `cuco`, `kvikio` even though we know other downstream projects use it.

It looks like to me that `include/cudf/interop.hpp` has been setup so that dlpack is a not required dependency and therefore we don't need to install it since not all consumers of libcudf need it.

Now if we do decide that we should install dlpack, the approach of installing `DLPACK_INCLUDE_DIR` is insufficient. dlpack has CMake install rules that packages up both the include directory and the `dlpack-config.cmake` module so others can use `find_package(dlpack)`. That is what we should be doing, and that logic would go into `get_dlpack.cmake`

Reply:
`dlpack.h` is referenced in [`interop.pxd`](https://github.com/rapidsai/cudf/blob/e321bf1507980e0b2e0f369e5ac8269dea2be6b3/python/cudf/cudf/_lib/cpp/interop.pxd#L14-L16), so it has to be available to the Cython wheel builds.

I assume dlpack isn't installed alongisde libcudf is because we're passing `DOWNLOAD_ONLY TRUE` in the CPM args. I will test removing this argument and configuring dlpack like any other dependency.

Reply:
The changes in https://github.com/rapidsai/cudf/pull/12139 also ensures that our python cudf builds will have dlpack if we want to just continue using that approach

Reply:
@robertmaynard that only works if libcudf is built as part of building the Python wheel, which in this case it isn't. It's built and installed separately, then the Python wheel is built standalone.

Reply:
Good point, I hadn't thought about wheel builds